### PR TITLE
feat(date-picker): switch between selection modes

### PR DESCRIPTION
**Pull request by imkesin**
## Description

These changes help standardize filtering in a date component.

Screenshot 2024-10-21 at 3 00 56 PM

### CODE REVIEW

File: src/components/DatePicker/ModeSelector/DatePickerModeSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+type RangePickerMode = 'dayRangePicker' | 'monthRangePicker'
+
+export type DatePickerMode = SingularPickerMode | RangePickerMode
+
+const DATE_RANGE_MODE_CONFIG: Record<DatePickerMode, { label: string }> = {
+  timePicker: { label: 'Time' },
+  dayPicker: { label: 'Day' },
+  dayRangePicker: { label: 'Custom Date Range' },
+  monthPicker: { label: 'Month' },
+  monthRangePicker: { label: 'Custom Month Range' },
```
Comment (doneel) [inline src/components/DatePicker/ModeSelector/DatePickerModeSelector.tsx:15]:
```suggestion
  monthRangePicker: { label: 'Select months' },
```

File: src/components/DatePicker/ModeSelector/DatePickerModeSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+type SingularPickerMode = 'dayPicker' | 'monthPicker' | 'timePicker'
+type RangePickerMode = 'dayRangePicker' | 'monthRangePicker'
+
+export type DatePickerMode = SingularPickerMode | RangePickerMode
+
+const DATE_RANGE_MODE_CONFIG: Record<DatePickerMode, { label: string }> = {
+  timePicker: { label: 'Time' },
+  dayPicker: { label: 'Day' },
+  dayRangePicker: { label: 'Custom Date Range' },
```
Comment (doneel) [inline src/components/DatePicker/ModeSelector/DatePickerModeSelector.tsx:13]:
```suggestion
  dayRangePicker: { label: 'Select dates' },
```

File: src/components/StatementOfCashFlow/StatementOfCashFlow.tsx [hunk 3]
```diff
@@ -88,6 +109,11 @@ const StatementOfCashFlowView = ({
         }
         dateFormat='MMM d'
         mode={datePickerMode}
+        allowedModes={['dayRangePicker', 'monthPicker']}
```
Comment (doneel) [inline src/components/StatementOfCashFlow/StatementOfCashFlow.tsx:112]: Minor: I'd love to swap the order here so it defaults to the first one and the 2nd looks like an optional advanced mode, which it is.

**Review by doneel — APPROVED** [review]

File: src/components/StatementOfCashFlow/StatementOfCashFlow.tsx [hunk 4]
```diff
@@ -79,6 +95,11 @@ const StatementOfCashFlowView = ({
         }}
         dateFormat='MMM'
         mode={datePickerMode}
+        allowedModes={['dayRangePicker', 'monthPicker']}
```
[the remark this answers is not in the document]
Comment (imkesin) [reply src/components/StatementOfCashFlow/StatementOfCashFlow.tsx:98]: We decided that this needs a bit more thought.

File: src/components/DatePicker/ModeSelector/DatePickerModeSelector.tsx (hunk 2 above)
Comment (imkesin) [reply src/components/DatePicker/ModeSelector/DatePickerModeSelector.tsx:13]: Are we default sentence case in most of the app?

**Pull request merged**